src/web/aws-cloudwatch/hooks/fetch.js

```diff
@@ -0,0 +1,81 @@
+import { useContext, useEffect, useState } from 'react';
+
+import URLUtils from 'util/URLUtils';
+import fetch from 'logic/rest/FetchProvider';
+
+import { FormDataContext } from '../context/FormData';
+import { awsAuth } from '../context/default_settings';
+
+/* useFetch Custom Hook
+
+Because of [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html) we have to get fancy
+in order to use fetch Promises and hooks.
+
+PARAMETERS:
+ - `setHook` that you want to call after fetch [required]
+ - `method`: '', [`'GET'` optional]
+ - `options`: '', [`{}` optional]
+ - `callback` that will fire any part of your code you need after fetch [`() => {}` optional]
+
+USE:
+`const [status, setUrl] = useFetch(setNextHook, onFooBar);`
+ - `status` will provide the current reducer state `{ loading, error, data }`
+ - `setUrl` will be your hook to call as a submit handler within a subfunction, it needs the API route as a string:
+
+EXAMPLES:
+ - See uses in `src/web/aws-cloudwatch/StepAuthorize.jsx`
+*/
+
+const useFetch = (setHook, method = 'GET', options = {}, callback = () => {}) => {
```

Comment:
Two things regarding the call signature, my proposal is this:

```
const useFetch = ({ method = 'GET', url = undefined, options = {} }, callback = () => {}) => {
  [...]
  const [fetchUrl, setFetchUrl] = useState(url);
```

  1.) I think we can collapse the `setHook` and the `callback`, they are doing the same, from `useFetch`'s perspective. If we want to have both a setter and a callback, we can call them sequentially from an anonymous function. (`callback` would get the state as parameter in this case)

  2.) If we are passing the `url` parameter when initializing `useFetch` (instead of through the returned setter), we can immediately do the fetch (instead of having to wrap the setter call in `useEffect` clause, like in `StepAuthorize`)